src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatReader.cs

```diff
@@ -74,6 +99,8 @@ private void ReflectionReadMembers(object obj, XmlReaderDelegator xmlReader, Xml
             bool hasRequiredMembers = (firstRequiredMember < memberCount);
             int requiredIndex = hasRequiredMembers ? firstRequiredMember : -1;
             int index = -1;
+            DataMember[] members = new DataMember[memberCount];
+            ReflectionGetMembers(_classContract, members);
```

Comment:
The value returned by ReflectionGetMembers should always equal to memberCount. I will do an assert here.